REFERENCE.md

```diff
@@ -0,0 +1,90 @@
+---
+RFC: <PR number>
+Status: <Proposed>
+---
+
+# Title (Required)
+
+## Abstract (Required)
+
+A few sentences describing the feature.
+
+## Motivation (Required)
+
+What the feature solves and why the existing functionality is not enough.
+
+## Design considerations (Required)
+
+A description of the design constraints and requirements for the proposal. Comparisons with similar features in other projects.
+
+## Specification (Required)
+
+A more detailed description of the feature, including why certain details in the design have been chosen.
+
+### Commands (Optional)
+
+If any new commands are introduced:
+
+1. Command name
+   - **Request**
+   - **Response**
+
+### Authentication and Authorization (Optional)
+
+If there are any changes around introducing new ACL command/categories for user access control.
+
+### Append-only file (Optional)
+
+If there are any changes around the persistence mechanism of every write operation.
+
+### Configuration (Optional)
+
+If there are any configuration changes introduced to enable/disable/modify the behavior of the feature.
+
+### Keyspace notifications (Optional)
+
+If there are any events to be introduced or modified to observe activity around the dataset.
+
+### Cluster mode (Optional)
+
+If there is any special handling for this feature (e.g., client redirection, Sharded PubSub, etc) in cluster mode, list out the behavior.
```

Comment:
I think it would be worthwhile calling out the changes to cluster messages as well, including new extensions, and new types, etc. We have seen compat issues in this area in the past.